tmd/io/io.py

```diff
@@ -54,8 +54,9 @@ def load_neuron(input_file, line_delimiter='\n', soma_type=None,
     that there are disconnected components
     '''
 
+    tree_types_final = td.copy()
     if tree_types is not None:
-        td.update(tree_types)
```

Comment:
This is a serious bug. Each time `load_neuron` is called with the `tree_types` argument it modifies the td dictionary. And the modification will persist between each call of `load_neuron`
You might have loaded wrong section types with realizing it.

Reply:
Ouch, good catch! The objective was to modify the tree_types for the lifetime of the load_neuron. Thanks for fixing this one!